joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/astcreation/AstForDeclarationsCreator.scala

```diff
@@ -233,12 +245,24 @@ trait AstForDeclarationsCreator {
     key: BabelNodeInfo,
     localTmpName: String
   ): Ast = {
-    val lhsElement = element.json("left")
     val rhsElement = element.json("right")
-    val lhsAst     = astForNode(lhsElement)
-    Ast.storeInDiffGraph(lhsAst, diffGraph)
-    val rhsAst = astForNodeWithFunctionReference(rhsElement)
+    val rhsAst     = astForNodeWithFunctionReference(rhsElement)
     Ast.storeInDiffGraph(rhsAst, diffGraph)
+
+    val lhsElement = element.json("left")
+    val lhsAst = createBabelNodeInfo(lhsElement) match {
+      case objPattern @ BabelNodeInfo(BabelAst.ObjectPattern) =>
+        val sourceAst = astForNodeWithFunctionReference(createBabelNodeInfo(rhsElement).json)
+        Ast.storeInDiffGraph(sourceAst, diffGraph)
```

Comment:
how about moving this to smth like store(ast)?

Reply:
Yeah, its already a one-line. I leave it as is for now.

Reply:
alrighty